spark/src/test/java/org/apache/iceberg/spark/data/TestHelpers.java

```diff
@@ -268,17 +268,18 @@ private static void assertEqualsUnsafe(Type type, Object expected, Object actual
 
     switch (type.typeId()) {
       case LONG:
+        Assert.assertTrue("Should be a long", actual instanceof Long);
         if (expected instanceof Integer) {
-          Assert.assertEquals("Values didn't match", ((Number) expected).longValue(),
-              ((Number) actual).longValue());
+          Assert.assertEquals("Values didn't match", ((Number) expected).longValue(), actual);
         } else {
           Assert.assertEquals("Primitive value should be equal to expected", expected, actual);
         }
         break;
       case DOUBLE:
+        Assert.assertTrue("Should be a double", actual instanceof Double);
         if (expected instanceof Float) {
           Assert.assertEquals("Values didn't match", Float.floatToIntBits(((Number) expected).floatValue()),
-              Float.floatToIntBits(((Number) actual).floatValue()));
+                    Float.floatToIntBits(((Number) actual).floatValue()));
```

Comment:
Why isn't this assertion like the one for `long`? Shouldn't this use the actual value and convert the expect to a `double`?

Reply:
Ugh..I just can't seem to get this right :-(.